public/locations-list-map.js

```diff
@@ -278,125 +377,103 @@ window.onFilterChange = function (elem, scrollNeeded) {
-     }
+
+
+/**
+ * Called once Google Maps API is fully loaded ready to run.
+ *
+ * IMPORTANT: This is exposed via window.initMap so the script can execute it.
+ */
+function initMap() {
+  var data_by_location = window.data_by_location;
+
+  var element = document.getElementById('map');
+
+  if (element == null) {
+    alert('could not find map div');
+  }
+
+  $(".map-container").show();
+
+  const singleStateFilter = showStates && showStates.length === 1;
+  const firstState = showStates[0];
+
+  // Load map at our initial center + zoom level.
+  map = new google.maps.Map(element, {zoom: initial_zoom, center: inital_center});
+
+  // filter states if there is a state filter
+  const filteredStates = Object.keys(data_by_location).filter((stateCode) => (
+    (
+      singleStateFilter
+      && firstState.toUpperCase() === stateCode.toUpperCase()
+      && stateLocationMappings[stateCode.toUpperCase()]
+      // add markers when no single state filter or if state code is invalid
+    ) || (!singleStateFilter || !stateLocationMappings[firstState.toUpperCase()])
+  ));
+
+  for (const state of filteredStates.sort()) {
+    const cities = data_by_location[state];
+
+    for (const city of Object.keys(cities).sort()) {
+      for (const entry of cities[city]) {
+        const name = entry["name"];
+        const address = entry["address"];
+        const latitude = entry["lat"];
+        const longitude = entry["lng"];
+        const instructions = entry["instructions"];
+        const accepting = entry["accepting"];
+        const open_accepted = entry["open_box"];
+        // Convert the lat and lng fields to numbers
+        if (!isNaN(Number(latitude))) {
+          var marker = addMarkerToMap(Number(latitude), Number(longitude),
+            address, name, instructions, accepting, open_accepted);
+          markers.push(marker);
+        }
+      }
+    }
+  }
+
+  if (singleStateFilter) {
+    // Center the map to a state if only one is set in query params
+    centerMapToState(firstState);
+  } else {
+    // Center the map on the nearest markers to the user if possible
+    centerMapToMarkersNearUser();
+  }
+
+
+  // Initialize the map search autocompleter.
+  autocomplete = new google.maps.places.Autocomplete(
+    document.getElementById('map-search'), {types: ['geocode']}
+  );
+
+  // Avoid paying for data that you don't need by restricting the set of
+  // place fields that are returned to just the address components.
+  autocomplete.setFields(['geometry']);
+
+  // When the user selects an address from the drop-down, populate the
+  // address fields in the form.
+  autocomplete.addListener('place_changed', loadSearchInMap);
+
+  // Setup event listeners for map action links.
+  $('#use-location').on('click', function (e) {
+    e.preventDefault();
+    centerMapToMarkersNearUser();
+  });
+
+  $('#reset-map').on('click', function (e) {
+    e.preventDefault();
+    map.setCenter(inital_center);
+    map.setZoom(initial_zoom);
```

Comment:
That's what the `initial_` vars are for, actually. Ugh; massive refactor (why I had to clean up so much already), need to have things properly centralized in order to augment with more functionality like this. I'll see if this can be improved.

Reply:
ok done!

![2020-03-25_09-36-54](https://user-images.githubusercontent.com/4269377/77561619-2ee54980-6e7c-11ea-995d-60169521dbd2.gif)